### PR TITLE
Update the plugin to insist on a build

### PR DESCRIPTION
**Pull request by hmgibson23**
Bundles the Apache License, adds necessary dependencies and makes opinionated choices about build systems.

\cc @cutandpastey

### CODE REVIEW

File: root/example/index.html [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
 
+<body>
+<h1>{%= name %}</h1>
+<div class='scribe'></div>
+<script src='../bower_components/requirejs/require.js'></script>
+<script>
+  require.config({
+    baseUrl: '../',
+    paths: {
+        'scribe': 'bower_components/scribe/scribe',
```
Comment (cutandpastey) [inline root/example/index.html]: This needs to be changed as the `bower.json` requires `scribe: 1.0.0` not `scribe: master`

File: root/example/index.html [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
+        '{%= name %}': "build/{%= name %}",
+    }
+  });
+  require(['scribe','{%= name %}'],
+          function(Scribe, scribePlugin()) {
```
Comment (cutandpastey) [inline root/example/index.html]: 2nd argument is not a function invocation

File: root/package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        "node": "{%= node_version %}"
+    },
+    "scripts": {
+        "test-firefox": "TEST_SERVER_PORT=9090 BROWSER_NAME='firefox' ./run-tests.sh",
+        "test-chrome": "TEST_SERVER_PORT=9090 BROWSER_NAME='chrome' ./run-tests.sh",
+        "test": "npm run test-chrome",
+        "postinstall": "bower install",
+        "build": "browserify src/{%= name %} --standalone {%= name %} -t babelify > ./build/{%= name %}.js",
+        "build-prod": "npm run build && uglifyjs ./build/{%= name %}.js > ./build/{%= name %}.min.js"
+    },
```
Comment (cutandpastey) [inline root/package.json:35]: Watchify?

File: root/test/app/index.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+<script src='../../bower_components/requirejs/require.js'></script>
+<script>
+  require.config({
+    baseUrl: '../../',
+    paths: {
+      'scribe': 'bower_components/scribe/scribe',
+       '{%= name %}': "build/{%= name %}"
+    }
+  });
+  require(['scribe','{%= name %}'],
```
Comment (cutandpastey) [inline root/test/app/index.html]: This isn't needed as its done in `loadPlugin`

File: root/test/app/index.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+<body>
+<h1>{%= name %}</h1>
+<div class='scribe'></div>
+<script src='../../bower_components/requirejs/require.js'></script>
+<script>
+  require.config({
+    baseUrl: '../../',
+    paths: {
+      'scribe': 'bower_components/scribe/scribe',
```
Comment (cutandpastey) [inline root/test/app/index.html]: Same deal with paths here and scribe.

File: root/test/helpers/load-plugin.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+var driver;
+
+before(function(){
+  driver = helpers.driver;
+});
+
+module.exports = function loadPlugin() {
+  return driver.executeAsyncScript(function (done) {
+    require(['scribe', '../../build/{%= name %}'], function (Scribe, plugin) {
+      window.scribe.use(plugin());
```
Comment (cutandpastey) [inline root/test/helpers/load-plugin.js:11]: Maybe add an object here and comment that shows its the plugin config object?

File: root/package.json (hunk 3 above)
Comment (hmgibson23) [inline root/package.json:35]: Yeah could do ... but then people might not want to do that.

File: root/test/helpers/load-plugin.js (hunk 6 above)
Comment (hmgibson23) [inline root/test/helpers/load-plugin.js:11]: Not sure what you mean about that.

**Comment by cutandpastey** [comment]
:+1:

**Comment by theefer** [comment]
Don't we want to move toward jspm rather than browserify & co?

**Comment by cutandpastey** [comment]
I think we should distribute the sources so you can load things with jspm.  We should also publish the built sources as we are good people.

**Comment by theefer** [comment]
OK! we can always review the situation once Composer and others actually do use jspm.

**Pull request merged**